src/constraints/energy.jl

```diff
@@ -7,42 +7,57 @@ Adds the energy constraints for assets withnin the period blocks of the timefram
 """
 
 function add_energy_constraints!(model, constraints, graph)
-
     ## INTER-TEMPORAL CONSTRAINTS (between representative periods)
 
-    # - Maximum outgoing energy within each period block
-    model[:max_energy_over_clustered_year] = [
-        @constraint(
+    let table_name = :max_energy_over_clustered_year, cons = constraints[table_name]
+        # - Maximum outgoing energy within each period block
+        attach_constraint!(
             model,
-            constraints[:max_energy_over_clustered_year].expressions[:outgoing][row.index] ≤
-            profile_aggregation(
-                sum,
-                graph[row.asset].timeframe_profiles,
-                row.year,
-                row.year,
-                "max_energy",
-                row.period_block_start:row.period_block_end,
-                1.0,
-            ) * (graph[row.asset].max_energy_timeframe_partition[row.year]),
-            base_name = "max_energy_over_clustered_year_limit[$(row.asset),$(row.year),$(row.period_block_start):$(row.period_block_end)]"
-        ) for row in eachrow(constraints[:max_energy_over_clustered_year].indices)
-    ]
+            cons,
+            table_name,
+            [
+                @constraint(
+                    model,
+                    outgoing_flow ≤
+                    profile_aggregation(
+                        sum,
+                        graph[row.asset].timeframe_profiles,
+                        row.year,
+                        row.year,
+                        "max_energy",
+                        row.period_block_start:row.period_block_end,
+                        1.0,
+                    ) * (graph[row.asset].max_energy_timeframe_partition[row.year]),
```

Comment:
I mean that the profile aggregation is done in an expression and not in the constraint. Like the ones in the capacity file (i.e., `constraints[table_name].expressions[:profile_times_capacity]`), which have the profile aggregation x something. But, here, the "something" is not a variable but a constant. Still, if you look at the constraints in the capacity file and then at these, you see a difference...but, as I said, maybe we should be clear in future developers' docs that expressions like `constraints[table_name].expressions[:ABC]` are only used when you have variables involved. Did I explain myself better this time? Sorry for the mess.

Reply:
Ah yeah. I see the `profile_times_capacity` as incidentally being in the expressions for the capacity.jl file. Profile stuff in general is a bit "undecided" as it is. Also because we're trying to remove it from `graph`, things might change heavily.
But indeed, if it doesn't contain any JuMP variable or expression, then creating an expression will lead to a different data type that can't be stored in the expressions object.

There probably should be some discussion on when to create expressions, and when to attach them, or just directly use. Many expressions are related to multi-year, so compact is at the core of the discussion

Reply:
Great! Thanks for the explanation and I agree we need to have a more detailed discussion about the expressions and try to unify their structure as much as possible.